spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AppConfigurationBootstrapConfiguration.java

```diff
@@ -67,31 +66,50 @@ public CloseableHttpClient closeableHttpClient() {
+            Optional<SecretClientBuilderSetup> keyVaultClientProviderOptional) {
+        
         KeyVaultCredentialProvider keyVaultCredentialProvider = null;
-        try {
-            keyVaultCredentialProvider = context.getBean(KeyVaultCredentialProvider.class);
-        } catch (NoUniqueBeanDefinitionException e) {
-            throw new RuntimeException("Failed to find unique KeyVaultCredentialProvider Bean for authentication.", e);
-        } catch (NoSuchBeanDefinitionException e) {
+        SecretClientBuilderSetup keyVaultClientProvider = null;
+        
+        if (!keyVaultCredentialProviderOptional.isPresent()) {
             LOGGER.debug("No KeyVaultCredentialProvider found.");
+        } else {
+           keyVaultCredentialProvider = keyVaultCredentialProviderOptional.get(); 
         }
+
+        if (!keyVaultClientProviderOptional.isPresent()) {
+            LOGGER.debug("No KeyVaultCredentialProvider found.");
+        } else {
+           keyVaultClientProvider = keyVaultClientProviderOptional.get(); 
+        }
+        
         return new AppConfigurationPropertySourceLocator(properties, appProperties, clients,
-                keyVaultCredentialProvider);
+                keyVaultCredentialProvider, keyVaultClientProvider);
     }
 
     @Bean
     public ClientStore buildClientStores(AppConfigurationProperties properties,
-            AppConfigurationProviderProperties appProperties, ConnectionPool pool, ApplicationContext context) {
+            AppConfigurationProviderProperties appProperties, ConnectionPool pool, ApplicationContext context,
+            Optional<AppConfigurationCredentialProvider> tokenCredentialProviderOptional,
+            Optional<ConfigurationClientBuilderSetup> clientProviderOptional) {
+        
         AppConfigurationCredentialProvider tokenCredentialProvider = null;
-        try {
-            tokenCredentialProvider = context.getBean(AppConfigurationCredentialProvider.class);
-        } catch (NoUniqueBeanDefinitionException e) {
-            throw new RuntimeException(
-                    "Failed to find unique AppConfigurationCredentialProvider Bean for authentication.", e);
-        } catch (NoSuchBeanDefinitionException e) {
+        ConfigurationClientBuilderSetup clientProvider = null;
+        
+        if (!tokenCredentialProviderOptional.isPresent()) {
             LOGGER.debug("No AppConfigurationCredentialProvider found.");
+        } else {
+            tokenCredentialProvider = tokenCredentialProviderOptional.get();
+        }
+        
+        if (!clientProviderOptional.isPresent()) {
+            LOGGER.debug("No AppConfigurationClientProvider found.");
+        } else {
+            clientProvider = clientProviderOptional.get();
         }
-        return new ClientStore(appProperties, pool, tokenCredentialProvider);
+        
+        return new ClientStore(appProperties, pool, tokenCredentialProvider, clientProvider);
```

Comment:
One idea to avoid an intermediate variable it would be to use the `Optional.else` directly here
```suggestion
return new ClientStore(
                appProperties, 
                pool, 
                tokenCredentialProviderOptional.orElse(null), 
                clientProviderOptional.orElse(null)
        );
```